src/openvic-simulation/misc/SoundEffect.cpp

```
@@ -1,38 +1,48 @@
 #include "SoundEffect.hpp"
 
+#include "openvic-simulation/dataloader/Dataloader.hpp"
+
 using namespace OpenVic;
 using namespace OpenVic::NodeTools;
 
-SoundEffect::SoundEffect (
-	std::string_view new_identifier, std::string_view new_file, fixed_point_t new_volume
-) : HasIdentifier { new_identifier }, file { new_file }, volume { new_volume } {}
+SoundEffect::SoundEffect( //
+	std::string_view new_identifier, std::filesystem::path new_file, fixed_point_t new_volume
+)
+	: HasIdentifier { new_identifier }, file { new_file }, volume { new_volume } {}
+
+bool SoundEffectManager::_load_sound_define(Dataloader const& dataloader, std::string_view sfx_identifier, ast::NodeCPtr root) {
+	std::filesystem::path file {};
+	auto file_callback = [&dataloader, &file](std::string_view val) -> bool {
+		std::string lookup = StringUtils::append_string_views("sound/", val);
+		file = dataloader.lookup_file(lookup, false);
+		if (file.empty()) {
+			Logger::warning("Lookup for \"", lookup, "\" failed!");
+		}
+		return true;
+	};
 
-bool SoundEffectManager::_load_sound_define(std::string_view sfx_identifier, ast::NodeCPtr root) {
-	std::string_view file {};
 	fixed_point_t volume = 1;
 	bool ret = expect_dictionary_keys(
-		"file", ONE_EXACTLY, expect_string(assign_variable_callback(file)),
-		"volume", ZERO_OR_ONE,
-			expect_fixed_point(assign_variable_callback(volume))
+		"file", ONE_EXACTLY, expect_string(file_callback), //
+		"volume", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(volume)) //
 	)(root);
 
 	if (sfx_identifier.empty()) {
 		Logger::error("Invalid sound identifier - empty!");
 		return false;
 	}
 	if (file.empty()) {
-		Logger::error("Invalid sound file name - empty!");
-		return false;
+		Logger::warning("Sound filename ", sfx_identifier, " was empty!");
```

Maybe specify lookup failed rather than saying the filename was empty? If you cache `std::string_view val` from `file_callback` you could include that in this error message

---

The callback already reports that, there is technically the capability for it to be empty for other reasons.